contracts/src/plugin/extensions/proposal/Proposal.sol

```diff
@@ -2,82 +2,35 @@
-    function _createProposalId() internal returns (uint256 proposalId) {
-        proposalId = proposalCount();
-        proposalCounter.increment();
-    }
-
-    /// @notice Internal function to create a proposal.
-    /// @param _metadata The proposal metadata.
-    /// @param _startDate The start date of the proposal in seconds.
-    /// @param _endDate The end date of the proposal in seconds.
-    /// @param _allowFailureMap A bitmap allowing the proposal to succeed, even if individual actions might revert. If the bit at index `i` is 1, the proposal succeeds even if the `i`th action reverts. A failure map value of 0 requires every action to not revert.
-    /// @param _actions The actions that will be executed after the proposal passes.
-    /// @return proposalId The ID of the proposal.
-    function _createProposal(
-        address _creator,
-        bytes calldata _metadata,
-        uint64 _startDate,
-        uint64 _endDate,
-        IDAO.Action[] calldata _actions,
-        uint256 _allowFailureMap
-    ) internal virtual returns (uint256 proposalId) {
-        proposalId = _createProposalId();
-
-        emit ProposalCreated({
-            proposalId: proposalId,
-            creator: _creator,
-            metadata: _metadata,
-            startDate: _startDate,
-            endDate: _endDate,
-            actions: _actions,
-            allowFailureMap: _allowFailureMap
-        });
-    }
-
-    /// @notice Internal function to execute a proposal.
-    /// @param _proposalId The ID of the proposal to be executed.
-    /// @param _actions The array of actions to be executed.
-    /// @param _allowFailureMap A bitmap allowing the proposal to succeed, even if individual actions might revert. If the bit at index `i` is 1, the proposal succeeds even if the `i`th action reverts. A failure map value of 0 requires every action to not revert.
-    /// @return execResults The array with the results of the executed actions.
-    /// @return failureMap The failure map encoding which actions have failed.
-    function _executeProposal(
-        IDAO _dao,
-        uint256 _proposalId,
-        IDAO.Action[] memory _actions,
-        uint256 _allowFailureMap
-    ) internal virtual returns (bytes[] memory execResults, uint256 failureMap) {
-        (execResults, failureMap) = _dao.execute(bytes32(_proposalId), _actions, _allowFailureMap);
-        emit ProposalExecuted({proposalId: _proposalId});
+        // In addition to the current interfaceId, also support previous version of the interfaceId
+        // that did not include the following functions:
+        // `createProposal, canExecute, createProposalId, createProposalParamsABI`.
+        return
+            _interfaceId ==
+            type(IProposal).interfaceId ^
+                IProposal.createProposal.selector ^
+                IProposal.canExecute.selector ^
+                IProposal.createProposalId.selector ^
+                IProposal.createProposalParamsABI.selector ||
```

Comment:
Shouldn't this be a versioned `IProposalv1` and `IProposalv2` interface?
It makes little sense that we are checking 2 different subsets of one interface